```
args = (parent: "projects/python-docs-samples-tests/locations/US"
capacity_commitment {
  slot_count: 100
  plan: FLEX
}
,)
kwargs = {'metadata': [('x-goog-request-params', 'parent=projects/python-docs-samples-tests/locations/US'), ('x-goog-api-client', 'gl-python/3.6.13 grpc/1.39.0 gax/1.31.1 gapic/1.2.2')]}

    @six.wraps(callable_)
    def error_remapped_callable(*args, **kwargs):
        try:
            return callable_(*args, **kwargs)
        except grpc.RpcError as exc:
>           six.raise_from(exceptions.from_grpc_error(exc), exc)
E           google.api_core.exceptions.FailedPrecondition: 400 Google owned projects are not allowed to buy FLEX slots, please see go/bq-flex-restrictions for details.

.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:69: FailedPrecondition
- generated xml file: /tmpfs/src/github/python-bigquery-reservation/samples/snippets/sponge_log.xml -
```


I've requested FLEX slot usage in internal bug 196099709

--- a/samples/snippets/conftest.py
+++ b/samples/snippets/conftest.py
@@ -12,7 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import datetime
 import os
+import time
 
 from google.cloud.bigquery_reservation_v1.services import reservation_service
 from google.cloud.bigquery_reservation_v1.types import reservation as reservation_types
@@ -50,4 +52,9 @@ def capacity_commitment(location_path: str, reservation_client: reservation_serv
     commitment.plan = reservation_types.CapacityCommitment.CommitmentPlan.FLEX
     commitment = reservation_client.create_capacity_commitment(parent=location_path, capacity_commitment=commitment)
     yield commitment
+    # Commitments can only be removed after 1 minute.
+    now = datetime.datetime.now(datetime.timezone.utc)
+    delta = commitment.commitment_end_time - now
+    sleep_seconds = max(0, delta.total_seconds()) + 5
+    time.sleep(sleep_seconds)
     reservation_client.delete_capacity_commitment(name=commitment.name)

